crates/red_knot_python_semantic/src/types/signatures.rs

```diff
@@ -29,10 +29,10 @@ impl<'db> Signature<'db> {
     }
 
     /// Return a todo signature: (*args: Todo, **kwargs: Todo) -> Todo
-    pub(crate) fn todo() -> Self {
+    pub(crate) fn todo(reason: &'static str) -> Self {
```

Comment:
I think this is going to lead to compiler warnings on release builds about unused arguments; we might need to do something similar to the `allow()` here:

https://github.com/astral-sh/ruff/blob/3aa7ba31b135bfeeb73e009415e0debb4ca47b95/crates/red_knot_python_semantic/src/symbol.rs#L55-L58